### PR TITLE
Fix RuboCop offenses in obs_factory

### PR DESCRIPTION
**Pull request by eduardoj**

### CODE REVIEW

**Comment by eduardoj** [comment]
[Fix Webui::ObsFactory rubocop offenses](https://trello.com/c/wkkclHw7/785-fix-webuiobsfactory-rubocop-offenses)

File: src/api/app/presenters/obs_factory/staging_project_presenter.rb [hunk 1]
```diff
@@ -147,19 +147,19 @@ def testing_percentage
     # the lower the number, the earlier it is in the list - acceptable A first
     def sort_key
       main = case overall_state
-      when :acceptable then
+             when :acceptable then
         0
```
Comment (DavidKang) [inline src/api/app/presenters/obs_factory/staging_project_presenter.rb]: What happened here? :thinking:

Comment (eduardoj) [reply src/api/app/presenters/obs_factory/staging_project_presenter.rb]: @DavidKang 

> What happened here? :thinking:

EnforcedStyle is 'case', the default. Do you prefer 'end' EnforcedStyle?

Comment (DavidKang) [reply src/api/app/presenters/obs_factory/staging_project_presenter.rb]: @eduardoj , I'm fine with the cop, but the content of each `when` should be indented

File: src/api/app/models/obs_factory/distribution.rb [hunk 2]
```diff
@@ -209,13 +209,13 @@ def openqa_filter(project)
...
-        when /^SUSE:SLE-15-SP\d:GA/ then DistributionStrategySLE15.new
-        when /^SUSE:SLE-12-SP.*CASP\d*/ then DistributionStrategyCasp.new
+          when 'openSUSE:Factory' then DistributionStrategyFactory.new
+          when 'openSUSE:Factory:PowerPC' then DistributionStrategyFactoryPPC.new
+          when /^openSUSE:Leap:15\.*/ then DistributionStrategyOpenSUSELeap15.new
+          when /^SUSE:SLE-12-SP\d:GA/ then DistributionStrategySLE12SP1.new
+          when 'SUSE:SLE-15:GA' then DistributionStrategySLE15.new
+          when /^SUSE:SLE-15-SP\d:GA/ then DistributionStrategySLE15.new
+          when /^SUSE:SLE-12-SP.*CASP\d*/ then DistributionStrategyCasp.new
         else raise UnknownDistribution
```
Comment (dmarcoux) [inline src/api/app/models/obs_factory/distribution.rb]: `else` should be aligned with `when` too?

File: src/api/app/presenters/obs_factory/staging_project_presenter.rb (hunk 1 above)
Comment (eduardoj) [reply src/api/app/presenters/obs_factory/staging_project_presenter.rb]: Contents each 'when' indented. Please re-review again.

File: src/api/app/models/obs_factory/distribution.rb (hunk 2 above)
Comment (eduardoj) [reply src/api/app/models/obs_factory/distribution.rb]: Right. Aligned. Please re-review again.

**Comment by marcus-h** [comment]
Hmm PATCH 5 seems to screw the indentation:

On 2018-07-26 04:42:27 -0700, Eduardo J. wrote:


> Subject: [PATCH 5/7] Fix Layout/CaseIndentation RuboCop offenses

<SNIP>

> diff --git a/src/api/app/models/obs_factory/distribution.rb b/src/api/app/models/obs_factory/distribution.rb
> index 025ff6547e..8b7c2fb70a 100644
> --- a/src/api/app/models/obs_factory/distribution.rb
> +++ b/src/api/app/models/obs_factory/distribution.rb
> @@ -209,14 +209,14 @@ def openqa_filter(project)
>  
>      def distribution_strategy_for_project(project)
>        s = case project.name
> -        when 'openSUSE:Factory' then DistributionStrategyFactory.new
> -        when 'openSUSE:Factory:PowerPC' then DistributionStrategyFactoryPPC.new
> -        when /^openSUSE:Leap:15\.*/ then DistributionStrategyOpenSUSELeap15.new
> -        when /^SUSE:SLE-12-SP\d:GA/ then DistributionStrategySLE12SP1.new
> -        when 'SUSE:SLE-15:GA' then DistributionStrategySLE15.new
> -        when /^SUSE:SLE-15-SP\d:GA/ then DistributionStrategySLE15.new
> -        when /^SUSE:SLE-12-SP.*CASP\d*/ then DistributionStrategyCasp.new
> -        else raise UnknownDistribution
> +          when 'openSUSE:Factory' then DistributionStrategyFactory.new
> +          when 'openSUSE:Factory:PowerPC' then DistributionStrategyFactoryPPC.new
> +          when /^openSUSE:Leap:15\.*/ then DistributionStrategyOpenSUSELeap15.new
> +          when /^SUSE:SLE-12-SP\d:GA/ then DistributionStrategySLE12SP1.new
> +          when 'SUSE:SLE-15:GA' then DistributionStrategySLE15.new
> +          when /^SUSE:SLE-15-SP\d:GA/ then DistributionStrategySLE15.new
> +          when /^SUSE:SLE-12-SP.*CASP\d*/ then DistributionStrategyCasp.new
> +          else raise UnknownDistribution

(Hmm using 4 spaces looks strange to me, but if rubocop complains...)

> diff --git a/src/api/app/presenters/obs_factory/staging_project_presenter.rb b/src/api/app/presenters/obs_factory/staging_project_presenter.rb
> index ceb30e1722..6e799a0234 100644
> --- a/src/api/app/presenters/obs_factory/staging_project_presenter.rb
> +++ b/src/api/app/presenters/obs_factory/staging_project_presenter.rb
> @@ -147,24 +147,24 @@ def testing_percentage
>      # the lower the number, the earlier it is in the list - acceptable A first
>      def sort_key
>        main = case overall_state
> -      when :acceptable then
> -        0
> -      when :review then
> -        10000 - review_percentage * 100
> -      when :testing then
> -        20000 - testing_percentage * 100
> -      when :building then
> -        30000 - build_progress[:percentage] * 100
> -      when :failed then
> -        40000
> -      when :unacceptable then
> -        50000
> -      when :empty
> -        60000
> -      else
> -        Rails.logger.error "untracked #{overall_state}"
> -        return
> -      end
> +             when :acceptable then
> +               0
> +             when :review then
> +               10000 - review_percentage * 100
> +             when :testing then
> +               20000 - testing_percentage * 100
> +             when :building then
> +               30000 - build_progress[:percentage] * 100
> +             when :failed then
> +               40000
> +             when :unacceptable then
> +               50000
> +             when :empty
> +               60000
> +             else
> +               Rails.logger.error "untracked #{overall_state}"
> +               return
> +             end

This looks really ugly and the indentation is inconsistent with the
indentation from above (4 spaces vs. 7 spaces).

**Comment by eduardoj** [comment]
@marcus-h:

> Hmm PATCH 5 seems to screw the indentation:

We can still live with the rules currently applied by RuboCop if the `case` line is moved down:

```
foo =
  case ...
  when ...
  else ...
  end
```

Better?

**Comment by eduardoj** [comment]
@marcus-h:

> but it seems rubocop does not like it:)

Yep. What you expected is labelled in RuboCop with "bad for all styles" ( https://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Layout/CaseIndentation ).

**Pull request merged**